### PR TITLE
Bug fixes in (1) running with frac_grid=T and GFDL MP and (2) restarting with frac_grid=T

### PR DESCRIPTION
**Pull request by ShanSunNOAA**
The crash in running GFDL MP with frac_grid=T is fixed by using GFS_surface_composites.F90/meta in Moorthi's branch "SM_Oct102020".

Restart can reproduce after
-- Not setting fice to zero in order to leave lake ice untouched in atmos_model.F90
-- Bad water temperature is only filtered at initial time in FV3GFS_io.F90
-- Moorthi's CCPP update

Co-authored-with: Shrinivas Moorthi shrinivas.moorthi@noaa.gov
Co-authored-with: Denise Worthen Denise.Worthen@noaa.gov

This replaced PR #200

### CODE REVIEW

**Comment by climbfuji** [comment]
Associated PRs:

https://github.com/NCAR/ccpp-physics/pull/524
https://github.com/NOAA-EMC/fv3atm/pull/204
https://github.com/ufs-community/ufs-weather-model/pull/304

For regression testing information, see https://github.com/ufs-community/ufs-weather-model/pull/304.

**Review by SMoorthi-emc — APPROVED** [review]

File: atmos_model.F90
Comment (junwang-noaa) [inline atmos_model.F90:1767]: The line may set lake ice to be zero, can we confirm the impact of removing this line on lake ice?

File: io/FV3GFS_io.F90
Comment (junwang-noaa) [inline io/FV3GFS_io.F90:1083]: I am curious how many print lines we will get from the write statement here? Do we need them?

File: io/FV3GFS_io.F90
Comment (junwang-noaa) [inline io/FV3GFS_io.F90:1363]: May I ask why we need this if statement condition?

File: io/FV3GFS_io.F90
Comment (SMoorthi-emc) [reply io/FV3GFS_io.F90:1083]: Yes, I have confirmed in my own experiments that ice fractionas over lakes are maintained after this fix.
The motivation that lead to finding this bug was to understand why ice concentration over lakes vanished in the coupled runs while the initial conditon had non zero values.

File: atmos_model.F90
Comment (SMoorthi-emc) [reply atmos_model.F90:1767]: Yes, I have confirmed in my own experiments that ice fractionas over lakes are maintained after this fix.
The motivation that lead to finding this bug was to understand why ice concentration over lakes vanished in the coupled runs while the initial condition had non zero values

File: io/FV3GFS_io.F90
Comment (SMoorthi-emc) [reply io/FV3GFS_io.F90:1083]: Wrong place, please ignore the above comment.

File: io/FV3GFS_io.F90
Comment (SMoorthi-emc) [reply io/FV3GFS_io.F90:1363]: Because this will fix if the initial conditon contains unphysical values.

**Comment by ShanSunNOAA** [comment]
Hi Jun, this one is ok, as it is inside if
(IPD_Data(nb)%Sfcprop%oceanfrac(ix)>0.).  There was one line setting fice
to zero without any condition, which Moorthi had removed.

Thanks for checking,
-Shan

**Review by climbfuji — COMMENTED** [review]
Looks good to me. Will approve once ccpp-physics is merged, the submodule pointer is updated and the `.gitmodules` change is reversed.

File: io/FV3GFS_io.F90
Comment (junwang-noaa) [reply io/FV3GFS_io.F90:1083]: We may need to consider commenting these print lines in future commit as there might be too many lines in the log file.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by ShanSunNOAA** [comment]
Hi Jun,

Should I comment out these lines in my upcoming commit today?

Shan

**Comment by junwang-noaa** [comment]
We can do it in future commit, if you make changes in the code now, you may
have to run RT again.

**Comment by ShanSunNOAA** [comment]
Ok, I will commit as is.   -Shan

**Review by climbfuji — APPROVED** [review]

**Pull request merged**
